ci/setup-build-test-gdk.ps1

```diff
@@ -34,61 +50,56 @@ class TestSuite {
     }
 }
 
-[string] $test_repo_url = "git@github.com:improbable/UnrealGDKEngineNetTest.git"
-[string] $test_repo_relative_uproject_path = "Game\EngineNetTest.uproject"
-[string] $test_project_name = "NetworkTestProject"
-[string] $test_repo_branch = "master"
 [string] $user_gdk_settings = "$env:GDK_SETTINGS"
 [string] $user_cmd_line_args = "$env:TEST_ARGS"
 [string] $gdk_branch = "$env:BUILDKITE_BRANCH"
 
-# If the testing repo has a branch with the same name as the current branch, use that
-$testing_repo_heads = git ls-remote --heads $test_repo_url $gdk_branch
-if($testing_repo_heads -Match [Regex]::Escape("refs/heads/$gdk_branch")) {
-    $test_repo_branch = $gdk_branch
-}
+[TestProjectTarget] $gdk_test_project = [TestProjectTarget]::new("git@github.com:spatialos/UnrealGDKTestGyms.git", $gdk_branch, "Game\GDKTestGyms.uproject", "GDKTestGyms")
+[TestProjectTarget] $native_test_project = [TestProjectTarget]::new("git@github.com:improbable/UnrealGDKEngineNetTest.git", $gdk_branch, "Game\EngineNetTest.uproject", "NativeNetworkTestProject")
 
 # Allow overriding testing branch via environment variable
-if (Test-Path env:TEST_REPO_BRANCH) {
-    $test_repo_branch = $env:TEST_REPO_BRANCH
+if (Test-Path env:GDK_TEST_REPO_BRANCH) {
+    $gdk_test_project.test_repo_branch = $env:GDK_TEST_REPO_BRANCH
+}
+if (Test-Path env:NATIVE_TEST_REPO_BRANCH) {
+    $native_test_project.test_repo_branch = $env:NATIVE_TEST_REPO_BRANCH
 }
 
 $tests = @()
 
-# If building all configurations, use the test gyms, since the network testing project only compiles for the Editor configs
-# There are basically two situations here: either we are trying to run tests, in which case we use EngineNetTest
-# Or, we try different build targets, in which case we use UnrealGDKTestGyms
-if (Test-Path env:BUILD_ALL_CONFIGURATIONS) {
-    $test_repo_url = "git@github.com:spatialos/UnrealGDKTestGyms.git"
-    $test_repo_relative_uproject_path = "Game\GDKTestGyms.uproject"
-    $test_project_name = "GDKTestGyms"
-
-    $tests += [TestSuite]::new("$test_repo_url", "$test_repo_branch", "$test_repo_relative_uproject_path", "EmptyGym", "$test_project_name", "TestResults", "SpatialGDK.", "$user_gdk_settings", $True, "$user_cmd_line_args")
+if ((Test-Path env:TEST_CONFIG) -And ($env:TEST_CONFIG -eq "Native")) {
```

Comment:
Note: this was slightly refactored, and modified to run Loadbalanced tests appropriately in single-worker mode